test/autogen/oai/test_selection.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+import datasets
+from flaml import oai
+
+class TestExemplarSelector(unittest.TestCase):
+    '''
+    template: default or user-defined, if default, key_order is specific
+    method: existing (str) or user-defined (func) or None (return in the order of train_data)
+    '''
+    @classmethod
+    def setUpClass(cls):
+        seed = 41
+        cls.data = datasets.load_dataset("piqa")
+        cls.train_data = cls.data["train"].shuffle(seed=seed)
+        cls.test_data = cls.data["test"].shuffle(seed=seed)
+        cls.key_order = ["goal", "sol1", "sol2", "label"]
+        cls.exemplars = list(cls.train_data)[:8]
+        cls.context = list(cls.test_data)[0]
+
+    def test_case_existing_method_default_template(self):
+        # Most cases should use the default template and existing methods
+        prompt_fn = oai.ExemplarSelector.get_few_shot_template(self.exemplars, method="random", 
+                                                               method_params={"k": 3}, template_params={"key_order": self.key_order})
```

Comment:
In general, the field to predict in the context is not necessarily the last key. Also, the number of such fields is not necessarily 1. Make sure the `prompt_fn` does not depend on such assumptions.